src/tests/ftest/cart/test_group_np_cli.c

```diff
@@ -95,24 +154,28 @@ test_run(void)
 	if (test_g.t_hold)
 		sleep(test_g.t_hold_time);
 
-	for (i = 0; i < rank_list->rl_nr; i++) {
-		rank = rank_list->rl_ranks[i];
-
-		server_ep.ep_grp = grp;
-		server_ep.ep_rank = rank;
-		rc = crt_req_create(test_g.t_crt_ctx[0], &server_ep,
-				    CRT_PROTO_OPC(TEST_GROUP_BASE,
-				    TEST_GROUP_VER, 1), &rpc_req);
-		D_ASSERTF(rc == 0 && rpc_req != NULL,
-			  "crt_req_create() failed. "
-			  "rc: %d, rpc_req: %p\n", rc, rpc_req);
-		rc = crt_req_send(rpc_req, client_cb_common, NULL);
-		D_ASSERTF(rc == 0, "crt_req_send() failed. rc: %d\n", rc);
-
-		tc_sem_timedwait(&test_g.t_token_to_proceed, 61, __LINE__);
-	}
+	server_ep.ep_grp = grp;
 
+	// Shutdown one particular rank
+	if (test_g.t_verify_swim_status.rank >= 0) {
 
+		// Check swim status on all (remaining) ranks
+		for (i = 0; i < rank_list->rl_nr; i++) {
+			rank = rank_list->rl_ranks[i];
+			server_ep.ep_rank = rank;
```

Comment:
no need to use 'rank' variable; just do server_ep.ep_rank = rank_list->rl_ranks[i];